packages/react-native-web/src/exports/Image/index.js

```diff
@@ -94,61 +95,105 @@ function getFlatStyle(style, blurRadius, filterId) {
   return [flatStyle, resizeMode, _filter, tintColor];
 }
 
-function resolveAssetDimensions(source) {
-  if (typeof source === 'number') {
-    const { height, width } = getAssetByID(source);
-    return { height, width };
-  } else if (
-    source != null &&
-    !Array.isArray(source) &&
-    typeof source === 'object'
-  ) {
-    const { height, width } = source;
-    return { height, width };
-  }
+function resolveAssetDimensions(source: ImageSource) {
+  const { height, width } = source;
+  return { height, width };
 }
 
-function resolveAssetUri(source): ?string {
-  let uri = null;
```

Comment:
Updated this function to always return an object, the existing logic was just moved to separate helper functions
- `resolveNumericSoure`
- `resolveStringSource`
- `resolveObjectSource`
- `resolveSvgDataUriSource`